### PR TITLE
Rewrite Android documentation to fit Android Studio/Gradle and SQLDroid better

### PR DESCRIPTION
**Pull request by jhannes**
As per http://stackoverflow.com/a/23340334/27658

### CODE REVIEW

File: documentation/api/index.html [hunk 1]
```diff
@@ -88,22 +88,45 @@
...
+    // ...
+}</pre>
+
+    <p>3. Include the setup in your main activity onCreate or application onCreate:</p>
 
-                <p>Then inside your main activity add this:</p>
                 <pre class="prettyprint">import org.flywaydb.core.Flyway;
 import org.flywaydb.core.api.android.ContextHolder;
-import android.database.sqlite.SQLiteDatabase;
+import org.sqldroid.DroidDataSource;
```
Comment (axelfontaine) [inline documentation/api/index.html:124]: Why the change in Driver class? Is the old one broken?

File: _layouts/api.html [hunk 2]
```diff
@@ -9,6 +9,12 @@
         {% if page.pill == 'overview' %} class="active" {% endif %}>
         <a href="/documentation/api">Overview</a></li>
         <li
+        {% if page.pill == 'overview' %} class="active" {% endif %}>
+        <a href="/documentation/api#android">Android</a></li>
+        <li
+        {% if page.pill == 'overview' %} class="active" {% endif %}>
+        <a href="/documentation/api#spring">Android</a></li>
+        <li
```
Comment (axelfontaine) [inline _layouts/api.html:17]: This changes the 1-1 mapping between navigation and pages and breaks the consistency of the convention used for the rest of the site.

File: _layouts/documentation.html [hunk 3]
```diff
@@ -43,7 +43,13 @@
...
                     {% if page.menu == 'api' %} class="active" {% endif %}>
-                    <a href="/documentation/api">API (Java / Android)</a></li>
+                    <a href="/documentation/api">API</a></li>
+                    <li
+                    {% if page.menu == 'api' %} class="active" {% endif %}>
+                    <a href="/documentation/api#android">API (Android)</a></li>
+                    <li
+                    {% if page.menu == 'api' %} class="active" {% endif %}>
+                    <a href="/documentation/api#spring">API (Spring)</a></li>
                     <li
```
Comment (axelfontaine) [inline _layouts/documentation.html:53]: This changes the 1-1 mapping between navigation and pages and breaks the consistency of the convention used for the rest of the site.

File: documentation/api/index.html [hunk 4]
```diff
@@ -88,22 +88,45 @@
         instance (both locally and on AWS), Flyway will automatically use the JDBC url, user and password
         <a href="https://boxfuse.com/docs/databases#envvars">provided by Boxfuse</a>.</div>
 
-    <h2>Programmatic Configuration (Android)</h2>
+    <h2 id="android">Android Programmatic Configuration (Android Studio)</h2>
```
Comment (axelfontaine) [inline documentation/api/index.html:92]: Is this really Android Studio-specific?

**Comment by axelfontaine** [comment]
Hi Johannes,

Thank you very much for this and apologies for the late review. Please find my comments inline.

Cheers
Axel

File: documentation/api/index.html (hunk 4 above)
Comment (jhannes) [inline documentation/api/index.html:92]: Not really. But Android development these days is almost almost documented in terms of Android Studio. Technically, the description only relies on Gradle (because of the definition of deps). But there is an old Gradle-way and a new Gradle-way (which came with Android Studio). I think the need to define assets is different in the old way.

In short: the details in this description are necessary and sufficient for up-to-date newly created android projects, but details are different for older projects.

File: _layouts/documentation.html (hunk 3 above)
Comment (jhannes) [inline _layouts/documentation.html:53]: I was conflicted about this. I found myself searching at random for the android info many times, however. It would be nice if it was more directly accessible. 

For me, the Java API is the supremely important part of the doc. The other pages, I have barely read.

File: documentation/api/index.html (hunk 1 above)
Comment (jhannes) [inline documentation/api/index.html:124]: Both will work. I suggest this variant because it makes explicit the connection between the database and Flyway. It helped me explain to colleagues what is going on. Which variant to prefer depends on how you want to use SqlDroid further.

File: _layouts/api.html (hunk 2 above)
Comment (jhannes) [inline _layouts/api.html:17]: I think the API page is larger than the others, and it covers more variants.

NB: I just spotted a typo in my patch: the link to Spring  is labeled "Android"

**Comment by axelfontaine** [comment]
Thanks Johannes! I ended up pulling in the changes in content, without the changes in the navigation.

Cheers
Axel
